src/dataconnect/build.ts

```diff
@@ -1,15 +1,80 @@
 import { DataConnectEmulator } from "../emulator/dataconnectEmulator";
 import { Options } from "../options";
 import { FirebaseError } from "../error";
-import { prettify } from "./graphqlError";
+import * as experiments from "../experiments";
+import { promptOnce } from "../prompt";
+import * as utils from "../utils";
+import { prettify, prettifyWithWorkaround } from "./graphqlError";
 import { DeploymentMetadata } from "./types";
 
-export async function build(options: Options, configDir: string): Promise<DeploymentMetadata> {
-  const buildResult = await DataConnectEmulator.build({ configDir });
+export async function build(
+  options: Options,
+  configDir: string,
+  dryRun?: boolean,
+): Promise<DeploymentMetadata> {
+  const args: { configDir: string; projectId?: string } = { configDir };
+  if (experiments.isEnabled("fdcconnectorevolution") && options.projectId) {
+    process.env["DATA_CONNECT_PREVIEW"] = "conn_evolution";
+    args.projectId = options.projectId;
+  }
+  const buildResult = await DataConnectEmulator.build(args);
   if (buildResult?.errors?.length) {
-    throw new FirebaseError(
-      `There are errors in your schema and connector files:\n${buildResult.errors.map(prettify).join("\n")}`,
+    if (buildResult.errors.filter((w) => !w.extensions?.warningLevel).length) {
+      // Throw immediately if there are any build errors in the GraphQL schema or connectors.
+      throw new FirebaseError(
+        `There are errors in your schema and connector files:\n${buildResult.errors.map(prettify).join("\n")}`,
+      );
+    }
+    const interactiveAcks = buildResult.errors.filter(
+      (w) => w.extensions?.warningLevel && w.extensions?.warningLevel === "INTERACTIVE_ACK",
```

Comment:
This is redundant currently - can simplify to
```suggestion
      (w) => w.extensions?.warningLevel === "INTERACTIVE_ACK",
```
because undefined !=INTERACTIVE_ACK